### PR TITLE
Remove `defaultProps` and `propTypes`

### PR DESCRIPTION
**Pull request by amoore108**
Overall changes
======
- Removes `defaultProps` and `propTypes` from the codebase as these will become removed/deprecated in React 19+ https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops
- Makes use of the ES6 default value to try and replicate the same behaviour as `defaultProps` before (see the code example in the React docs linked above for a demonstration of this)
- Updates some tests, namely hooks, to use our custom React Testing Library component as some of the functionality in `testing-library/react-hooks` is becoming deprecated
- Adds a basic `Article` type to replace the `inferProps` version previously
- Removes `react-test-renderer` as it will become deprecated and is not used in the codebase

Code changes
======
- Removes `defaultProps` and `propTypes` declarations from the codebase in favour of using ES6 default values
- This does remove the prop-type type checking, which is _kinda_ similar to TypeScript. This means that older JSX components may not have their props typed anymore. Newer TSX components shouldn't be using prop-types, so they should remain unaffected. There is inference though, so most components will display their props if you CMD+Space or hover over the component. 
- Some snapshots have been updated as Emotion added in a blank space before the first classname in some instances. This was resolved by conditionally applying the `className` prop on affected components

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Comment by amoore108** [comment]
Not quite sure how to use ES6 syntax for styled-components like: https://github.com/bbc/simorgh/blob/12c0e603660e11029e9b7e6a17caff17e6b4ab9c/src/app/legacy/psammead/psammead-bulleted-list/src/index.jsx#L7-L39

May not be a problem if React 19 doesn't shout about it.

File: src/app/components/RelatedContentSection/index.tsx [hunk 1]
```diff
@@ -165,8 +165,7 @@ const RelatedContentSection = ({ content }: { content: OptimoBlock[] }) => {
       role="region"
       data-e2e={LABEL_ID}
     >
-      {/* // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-expect-error - TS inferring props it thinks are required  */}
+      {/* // eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
```
Comment (karinathomasbbc) [inline src/app/components/RelatedContentSection/index.tsx]: Can this linting override now be removed?

File: src/app/legacy/components/Grid/index.jsx [hunk 2]
```diff
@@ -149,9 +149,15 @@ export const GridItemSmall = styled(props => (
   ${layoutGridItemSmall}
 `;
 
-export const GridItemMedium = ({ gridColumnStart, gridSpan, ...props }) => (
+export const GridItemMedium = ({
+  gridColumnStart = 5,
+  gridSpan = 10,
+  as = undefined,
```
Comment (karinathomasbbc) [inline src/app/legacy/components/Grid/index.jsx]: Do we need to initialise `as` as `undefined` here, as omitting it / no default value is the same as `undefined`?

File: src/app/legacy/components/OptimoPromos/Content/index.jsx [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
-import { string, bool } from 'prop-types';
 import moment from 'moment';
 import formatDuration from '#app/lib/utilities/formatDuration';
 import VisuallyHiddenText from '../../../../components/VisuallyHiddenText';
 import { ServiceContext } from '../../../../contexts/ServiceContext';
 import PromoContext from '../PromoContext';
 
-const Content = ({ mediaDuration, headline, isPhotoGallery, isLive }) => {
+const Content = ({
+  mediaDuration = '',
```
Comment (karinathomasbbc) [inline src/app/legacy/components/OptimoPromos/Content/index.jsx:9]: The default value was originally `null` - does it matter that it's now empty string?

File: src/app/legacy/components/Promo/timestamp.jsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import React, { useContext } from 'react';
-import { oneOfType, number, string } from 'prop-types';
 import Timestamp from '#psammead/psammead-timestamp-container/src';
 import isTenHoursAgo from '#lib/utilities/isTenHoursAgo';
 import { ServiceContext } from '../../../contexts/ServiceContext';
 
-const PromoTimestamp = ({ children, serviceDatetimeLocale, className }) => {
+const PromoTimestamp = ({
+  children,
+  serviceDatetimeLocale = '',
```
Comment (karinathomasbbc) [inline src/app/legacy/components/Promo/timestamp.jsx:8]: Original default was `null`

File: src/app/legacy/components/Promo/timestamp.jsx [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
+  className = '',
```
Comment (karinathomasbbc) [inline src/app/legacy/components/Promo/timestamp.jsx:9]: Do you need to check whether `className` is defined before adding it as a prop somewhere below?

File: .eslintrc.js
Comment (karinathomasbbc) [inline .eslintrc.js:1]: Can we get rid of all / any prop type linting now? Or do we want to keep it in case someone accidentally reinstates it? (Which is very unlikely and stuff will break once we go to React 19). If we do keep, should we switch them on to ensure that we've covered everything?

```
        'react/prop-types': 'off',
        'react/require-default-props': 'off',
        'react/no-unused-prop-types': 'off',
```

File: src/app/legacy/containers/Brand/index.jsx [hunk 6]
```diff
@@ -16,7 +15,12 @@ const StyledBrand = styled(Brand)`
   }
 `;
 
-const BrandContainer = ({ skipLink, scriptLink, brandRef, ...props }) => {
+const BrandContainer = ({
+  skipLink = null,
+  scriptLink = null,
+  brandRef = null,
+  ...props
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/Brand/index.jsx:22]: Do we need `borderTop` and `borderBottom` here? Or is it OK since `undefined` === `false`?

```suggestion
const BrandContainer = ({
  skipLink = null,
  scriptLink = null,
  brandRef = null,
  borderTop: false,
  borderBottom: false,
  ...props
```

File: src/app/legacy/containers/PageHandlers/withContexts.jsx [hunk 7]
```diff
@@ -15,25 +11,25 @@ const WithContexts = Component => {
...
-      status,
-      id,
+      bbcOrigin = null,
+      status = null,
+      id = null,
       service,
       isAmp,
       isApp,
       isLite,
       isCaf,
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/PageHandlers/withContexts.jsx]: Do these also need to be defaulted to `false` per the `defaultProps` below?

```suggestion
      isLite = false,
      isCaf = false,
```

File: src/app/legacy/containers/PageHandlers/withData/index.jsx [hunk 8]
```diff
@@ -1,14 +1,12 @@
...
-import { pageDataPropType } from '#models/propTypes/data';
 import { ServiceContext } from '../../../../contexts/ServiceContext';
 import { RequestContext } from '../../../../contexts/RequestContext';
 import shouldRender from './shouldRender';
 
 const WithData = Component => {
-  const DataContainer = ({ pageData, status, ...props }) => {
-    const { service, location: { pathname } = {} } = props;
+  const DataContainer = ({ pageData = null, status, ...props }) => {
+    const { service = 'default', location: { pathname } = {} } = props;
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/PageHandlers/withData/index.jsx]: Should this be an actual service e.g. `news`?

File: src/app/legacy/containers/RadioSchedule/Canonical/index.jsx [hunk 9]
```diff
@@ -96,7 +94,12 @@ const RadioFrequencyLink = styled.a`
   }
 `;
 
-const CanonicalRadioSchedule = ({ initialData, endpoint, lang, className }) => {
+const CanonicalRadioSchedule = ({
+  initialData = undefined,
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/RadioSchedule/Canonical/index.jsx]: Is `undefined` required here?

File: src/app/legacy/containers/RelatedTopics/index.jsx [hunk 10]
```diff
@@ -32,12 +31,12 @@ const StyledSectionLabel = styled(SectionLabel)`
 `;
 
 const RelatedTopics = ({
-  topics,
-  mobileDivider,
-  bar,
-  className,
-  backgroundColour,
-  tagBackgroundColour,
+  topics = [],
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/RelatedTopics/index.jsx:34]: The original default was `null`

File: src/app/legacy/containers/RelatedTopics/index.jsx [hunk 11]
```diff
@@ -32,12 +31,12 @@ const StyledSectionLabel = styled(SectionLabel)`
...
+  mobileDivider = true,
+  bar = true,
+  className = '',
+  backgroundColour = '',
+  tagBackgroundColour = '',
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/RelatedTopics/index.jsx:39]: Originals were `null`

File: src/app/legacy/containers/StoryPromo/index.jsx [hunk 12]
```diff
@@ -36,20 +34,23 @@ import useCombinedClickTrackerHandler from './useCombinedClickTrackerHandler';
...
+  isAmp = false,
   useLargeImages,
-  imageValues,
+  imageValues = {
+    path: '',
+    altText: '',
+    height: '',
+    width: '',
+  },
   lazyLoad,
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/StoryPromo/index.jsx]: Original default of `lazyload` was `false`

File: src/app/legacy/psammead/psammead-brand/src/index.jsx [hunk 13]
```diff
@@ -119,7 +118,11 @@ const BrandSvg = styled.svg`
   }
 `;
 
-const LocalisedBrandName = ({ linkId, product, serviceLocalisedName }) => {
+const LocalisedBrandName = ({
+  linkId = null,
```
Comment (karinathomasbbc) [inline src/app/legacy/psammead/psammead-brand/src/index.jsx:122]: Does it matter that this is a required value (it did not have a default value before)?

File: src/app/legacy/psammead/psammead-image/src/index.jsx [hunk 14]
```diff
@@ -37,12 +36,12 @@ const StyledImg = styled.img`
 export const Img = props => {
   const {
     src,
-    srcset,
-    sizes,
-    fallbackSrcset,
-    primaryMimeType,
-    fallbackMimeType,
-    onLoad,
+    srcset = null,
```
Comment (karinathomasbbc) [inline src/app/legacy/psammead/psammead-image/src/index.jsx:39]: Do we need to set `fade` = false here?

File: src/app/legacy/psammead/psammead-timestamp-container/src/index.jsx [hunk 15]
```diff
@@ -11,17 +9,17 @@ import {
...
-  format,
-  isRelative,
-  padding,
-  prefix,
-  suffix,
-  timezone,
+  format = '',
+  isRelative = false,
+  padding = true,
+  prefix = '',
```
Comment (karinathomasbbc) [inline src/app/legacy/psammead/psammead-timestamp-container/src/index.jsx:15]: Should `prefix` be `null` (like `suffix`)?

**Review by karinathomasbbc — APPROVED** [review]
One question / concern - if we were to ever convert any of these components to Typescrpt, we don't have PropTypes to tell us if fields are required or not.

We will also need to update the Migrate Legacy Component doc to remove any mention of propTypes (if we haven't already done so)

File: src/app/components/MostRead/Canonical/LastUpdated/index.tsx [hunk 16]
```diff
@@ -30,6 +30,7 @@ const LastUpdated = ({
     locale={locale}
     timezone={timezone}
     padding={false}
+    altCalendar={null}
```
Comment (andrewscfc) [inline src/app/components/MostRead/Canonical/LastUpdated/index.tsx:33]: why was this introduced?

File: src/app/contexts/ServiceContext/__mocks__/index.tsx [hunk 17]
```diff
@@ -17,11 +16,12 @@ interface Props {
  * using service contexts.
  */
 export const ServiceContext = React.createContext({});
+
 export const ServiceContextProvider = ({
   children,
   pageLang,
```
Comment (andrewscfc) [inline src/app/contexts/ServiceContext/__mocks__/index.tsx]:
```suggestion
  pageLang = null,
```

I'm not sure if this is necessary? I imagine as writtent `pageLang` comes through as `undefined` instead of `null`. Probably doesn't matter though?

File: src/app/hooks/useMediaQuery/index.test.jsx [hunk 18]
```diff
@@ -36,9 +40,8 @@ describe('useMediaQuery', () => {
   it('should call removeListener on cleanup', () => {
     const testFn = jest.fn();
     renderHook(() => useMediaQuery('(max-width: 600px)', testFn));
-    cleanup().then(() => {
-      expect(mockRemoveListener.mock.calls.length).toEqual(1);
-      expect(mockRemoveListener.mock.calls[0][0]).toEqual(testFn);
-    });
+    cleanup();
```
Comment (andrewscfc) [inline src/app/hooks/useMediaQuery/index.test.jsx:43]:
```suggestion
    await cleanup();
```
Does cleanup return a promise, just wondered if you've intentionally changed this logic?

File: src/app/hooks/useOptimizelyScrollDepth/index.test.jsx [hunk 19]
```diff
@@ -65,21 +61,17 @@ describe('useOptimizelyScrollDepth', () => {
...
     renderHook(() => useOptimizelyScrollDepth());
 
-    cleanup().then(() => {
-      expect(removeEventListenerSpy).toHaveBeenCalledWith(
-        'scroll',
-        expect.any(Function),
-        { passive: true },
-      );
-    });
+    cleanup();
```
Comment (andrewscfc) [inline src/app/hooks/useOptimizelyScrollDepth/index.test.jsx:64]: same here re: async

File: src/app/hooks/useTrackingToggle/index.test.jsx [hunk 20]
```diff
@@ -9,15 +9,8 @@ const wrapper = ({ children, toggles }) => (
 
 const isEnabled = (componentName, enabled, value) => {
   const { result } = renderHook(() => useTrackingToggle(componentName), {
-    wrapper,
-    initialProps: {
```
Comment (andrewscfc) [inline src/app/hooks/useTrackingToggle/index.test.jsx:13]: Is `initialProps` part of propTypes?

File: src/app/legacy/components/MediaPlayer/Guidance/index.jsx [hunk 21]
```diff
@@ -72,14 +71,14 @@ const Guidance = ({
   service,
   noJsMessage,
   noJsClassName,
```
Comment (andrewscfc) [inline src/app/legacy/components/MediaPlayer/Guidance/index.jsx]: Same as above, should these explicitly be marked as `null` rather than `undefined`?

File: src/app/legacy/components/MediaPlayer/Placeholder/index.jsx [hunk 22]
```diff
@@ -55,8 +54,14 @@ const Placeholder = ({
...
   src,
-  srcset,
-  mediaInfo,
+  srcset = null,
+  mediaInfo = {
+    datetime: null,
+    duration: null,
+    durationSpoken: null,
+    type: 'video',
+    guidanceMessage: '',
```
Comment (andrewscfc) [inline src/app/legacy/components/MediaPlayer/Placeholder/index.jsx]: has this intentionally changed to `''` from `null`?

File: src/app/legacy/components/OptimoPromos/Content/index.jsx (hunk 3 above)
Comment (andrewscfc) [inline src/app/legacy/components/OptimoPromos/Content/index.jsx:9]: should this be null?

File: src/app/legacy/components/Promo/timestamp.jsx (hunk 5 above)
Comment (andrewscfc) [inline src/app/legacy/components/Promo/timestamp.jsx:9]: empty string rather than null?

File: src/app/legacy/containers/ArticleMetadata/index.jsx [hunk 23]
```diff
@@ -1,23 +1,22 @@
...
   author,
-  twitterHandle,
+  twitterHandle = null,
   firstPublished,
   lastPublished,
-  section,
-  aboutTags,
+  section = '',
+  aboutTags = [],
   mentionsTags,
```
Comment (andrewscfc) [inline src/app/legacy/containers/ArticleMetadata/index.jsx]:
```suggestion
  mentionsTags = [],
```

File: src/app/legacy/containers/ArticleTimestamp/index.jsx [hunk 24]
```diff
@@ -18,9 +17,9 @@ const LastUpdatedTimestamp = Timestamp;
 const ArticleTimestamp = ({
   firstPublished,
   lastPublished,
-  popOut,
-  minutesTolerance,
-  className,
+  popOut = true,
+  minutesTolerance = 0,
+  className = '',
```
Comment (andrewscfc) [inline src/app/legacy/containers/ArticleTimestamp/index.jsx:22]:
```suggestion
  className,
```
I can't see `className` in the default props?

File: src/app/legacy/containers/Brand/index.jsx [hunk 25]
```diff
@@ -16,7 +15,12 @@ const StyledBrand = styled(Brand)`
   }
 `;
 
-const BrandContainer = ({ skipLink, scriptLink, brandRef, ...props }) => {
+const BrandContainer = ({
+  skipLink = null,
+  scriptLink = null,
+  brandRef = null,
```
Comment (andrewscfc) [inline src/app/legacy/containers/Brand/index.jsx:21]:
```suggestion
  brandRef = null,
  borderTop: false,
  borderBottom: false,
```

File: src/app/legacy/containers/BulletedList/index.jsx [hunk 26]
```diff
@@ -21,7 +20,7 @@ const withClickHandler = (Component, clickHandler) => props => (
 
 const BulletedListContainer = ({
   blocks,
-  className,
+  className = '',
```
Comment (andrewscfc) [inline src/app/legacy/containers/BulletedList/index.jsx:23]:
```suggestion
  className = null,
```

**Review by pvaliani — APPROVED** [review]
Copypasta from Slack:

Nice! Thank you.

For the styled components issue with defaultProps, since React isn’t raising flags about their usage currently, it might be safe to leave them as is in the short term. However, for long-term maintenance and consistency, it might be worthwhile to schedule a gradual rewrite of these components into a supported pattern, possibly for when we tackle other refactoring tasks or feature upgrades.

Would it be worthwhile to add additional tests focusing on targeted components where defaultProps were replaced with default parameters to ensure there are no edge cases we're missing where the behaviour could diff?

File: src/app/legacy/containers/PageHandlers/withPageWrapper/index.jsx [hunk 27]
```diff
@@ -10,18 +8,6 @@ const WithPageWrapper = Component => {
     </PageWrapper>
   );
 
-  PageWrapperContainer.propTypes = {
-    pageData: pageDataPropType,
-    status: number,
-    bbcOrigin: string,
-  };
-
-  PageWrapperContainer.defaultProps = {
```
Comment (andrewscfc) [inline src/app/legacy/containers/PageHandlers/withPageWrapper/index.jsx:19]: should these props be on `PageWrapperContainer`?

File: src/app/legacy/containers/WebVitals/index.jsx [hunk 28]
```diff
@@ -44,12 +43,4 @@ const WebVitals = ({ pageType }) => {
   return null;
 };
 
-WebVitals.propTypes = {
-  pageType: string,
-};
-
-WebVitals.defaultProps = {
-  pageType: null,
```
Comment (andrewscfc) [inline src/app/legacy/containers/WebVitals/index.jsx:52]: should this be applied?

File: src/app/legacy/psammead/psammead-heading-index/src/index.jsx [hunk 29]
```diff
@@ -11,11 +9,6 @@ const HeadingIndex = styled.h1`
   margin: 0;
 `;
 
-HeadingIndex.propTypes = {
-  script: shape(scriptPropType).isRequired,
-  service: string.isRequired,
-};
-
 HeadingIndex.defaultProps = {
```
Comment (andrewscfc) [inline src/app/legacy/psammead/psammead-heading-index/src/index.jsx:12]: this has been missed

File: src/app/legacy/psammead/psammead-headings/src/index.jsx [hunk 30]
```diff
@@ -46,11 +39,6 @@ export const SubHeading = styled.h2`
   }
 `;
 
-SubHeading.propTypes = {
-  script: shape(scriptPropType).isRequired,
-  service: string.isRequired,
-};
-
 SubHeading.defaultProps = {
```
Comment (andrewscfc) [inline src/app/legacy/psammead/psammead-headings/src/index.jsx:42]: this has been missed

File: src/app/legacy/psammead/psammead-image/src/index.amp.jsx [hunk 31]
```diff
@@ -10,34 +9,16 @@ const omitInvalidProps = omit([
 ]);
 
 const AmpImg = props => {
-  const { srcset, fallbackSrcset, ...otherProps } = props;
+  const { srcset, fallbackSrcset, attribution = '', ...otherProps } = props;
```
Comment (andrewscfc) [inline src/app/legacy/psammead/psammead-image/src/index.amp.jsx:12]: should the new values be reflected here?

File: src/app/legacy/psammead/psammead-story-promo-list/src/index.jsx [hunk 32]
```diff
@@ -35,8 +29,6 @@ export const StoryPromoLiBase = styled.li`
   }
 `;
 
-StoryPromoLiBase.propTypes = promoListPropTypes;
-
 StoryPromoLiBase.defaultProps = {
```
Comment (andrewscfc) [inline src/app/legacy/psammead/psammead-story-promo-list/src/index.jsx:32]: have these been missed?

File: src/app/legacy/components/Promo/timestamp.jsx (hunk 5 above)
Comment (amoore108) [reply src/app/legacy/components/Promo/timestamp.jsx:9]: Not in this case as the prop ends up getting checked at the final child component.

File: .eslintrc.js
Comment (amoore108) [reply .eslintrc.js:1]: We need to keep these here for now as eslint will complain. We can likely remove them once React 19 and eslint is updated with the new rules.

File: src/app/legacy/containers/Brand/index.jsx (hunk 6 above)
Comment (amoore108) [reply src/app/legacy/containers/Brand/index.jsx:22]: They are initialised to `false` here: https://github.com/bbc/simorgh/blob/9a577693f5da22ad8a8ef391a1a588683888fb5c/src/app/legacy/psammead/psammead-brand/src/index.jsx#L167-L180

File: src/app/legacy/psammead/psammead-timestamp/src/index.jsx [hunk 33]
```diff
@@ -29,39 +27,23 @@ const StyledTimestamp = styled.time`
...
   children,
   datetime,
-  typographyFunc,
+  typographyFunc = getBrevier,
   script,
-  padding,
+  padding = true,
   service,
-  className,
+  className = '',
```
Comment (andrewscfc) [inline src/app/legacy/psammead/psammead-timestamp/src/index.jsx:34]:
```suggestion
  className = null,
```

File: src/app/legacy/containers/PageHandlers/withContexts.jsx (hunk 7 above)
Comment (amoore108) [reply src/app/legacy/containers/PageHandlers/withContexts.jsx]: Good spot, will add

File: src/app/lib/utilities/srcSet/index.js [hunk 34]
```diff
@@ -31,7 +31,7 @@ export const createSrcsets = ({
   imageResolutions = DEFAULT_RESOLUTIONS,
 }) => {
   if (originCode === 'pips') {
-    return { primarySrcset: null, fallbackSrcset: null };
+    return { primarySrcset: undefined, fallbackSrcset: undefined };
```
Comment (andrewscfc) [inline src/app/lib/utilities/srcSet/index.js:34]: why is this needed?

File: src/app/legacy/containers/PageHandlers/withData/index.jsx (hunk 8 above)
Comment (amoore108) [reply src/app/legacy/containers/PageHandlers/withData/index.jsx]: Will try setting it to `news`

File: src/app/pages/ArticlePage/ArticlePage.jsx [hunk 35]
```diff
@@ -271,13 +269,4 @@ const ArticlePage = ({ pageData }) => {
   );
 };
 
-ArticlePage.propTypes = {
-  pageData: articleDataPropTypes.isRequired,
-  mostReadEndpointOverride: string,
-};
-
-ArticlePage.defaultProps = {
-  mostReadEndpointOverride: null,
```
Comment (andrewscfc) [inline src/app/pages/ArticlePage/ArticlePage.jsx:280]: Does this need adding above?

File: src/app/legacy/containers/RadioSchedule/Canonical/index.jsx (hunk 9 above)
Comment (amoore108) [reply src/app/legacy/containers/RadioSchedule/Canonical/index.jsx]: Doesn't seem to be, will remove.

File: src/app/legacy/containers/RelatedTopics/index.jsx (hunk 10 above)
Comment (amoore108) [reply src/app/legacy/containers/RelatedTopics/index.jsx:34]: This is changed because the MediaArticlePage renders RelatedTopics and it expects an array of some kind rather than `null`.

File: src/app/legacy/containers/RelatedTopics/index.jsx (hunk 11 above)
Comment (amoore108) [reply src/app/legacy/containers/RelatedTopics/index.jsx:39]: Should have the same affect as these props are added only if they aren't falsy.

File: src/app/legacy/containers/StoryPromo/index.jsx (hunk 12 above)
Comment (amoore108) [reply src/app/legacy/containers/StoryPromo/index.jsx]: will change

**Review by andrewscfc — COMMENTED** [review]
Huge respect doing this presumably without a codemod! I'm not sure if all my points are necessary, just wanted to point out semantic differences so they can be checked

File: src/app/legacy/psammead/psammead-brand/src/index.jsx (hunk 13 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-brand/src/index.jsx:122]: Which value are you referencing? `linkId` had a default of `null` and `serviceLocalisedName` had a default of `null`

File: src/app/legacy/psammead/psammead-image/src/index.jsx (hunk 14 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-image/src/index.jsx:39]: `fade` doesn't appear to be dynamically set anywhere to `true` that I can see in that component, so I left out the default as it will retain the same behaviour as `false` if its omitted entirely.

File: src/app/legacy/psammead/psammead-timestamp-container/src/index.jsx (hunk 15 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-timestamp-container/src/index.jsx:15]: I'll set both to `''` as the TS component that is using this complains if `prefix` is set to `null`

File: src/app/legacy/components/Promo/timestamp.jsx (hunk 4 above)
Comment (amoore108) [reply src/app/legacy/components/Promo/timestamp.jsx:8]: Should retain the same behaviour as its falsy. This is to satisfy TS where components passing in this prop value are strings.

File: src/app/legacy/components/OptimoPromos/Content/index.jsx (hunk 3 above)
Comment (amoore108) [reply src/app/legacy/components/OptimoPromos/Content/index.jsx:9]: Shouldn't matter, should retain the same behaviour as its falsy

File: src/app/components/RelatedContentSection/index.tsx (hunk 1 above)
Comment (amoore108) [reply src/app/components/RelatedContentSection/index.tsx]: Good spot, will remove.

File: src/app/legacy/components/Grid/index.jsx (hunk 2 above)
Comment (amoore108) [reply src/app/legacy/components/Grid/index.jsx]: Yep thats right, I'll remove.

File: src/app/contexts/ServiceContext/__mocks__/index.tsx (hunk 17 above)
Comment (amoore108) [reply src/app/contexts/ServiceContext/__mocks__/index.tsx]: I'll add it in for safety

File: src/app/hooks/useMediaQuery/index.test.jsx (hunk 18 above)
Comment (amoore108) [reply src/app/hooks/useMediaQuery/index.test.jsx:43]: It doesn't return a promise from the `@testing-react/react` library, but did in the now 'old' `@testing-react/react-hooks` library

File: src/app/hooks/useOptimizelyScrollDepth/index.test.jsx (hunk 19 above)
Comment (amoore108) [reply src/app/hooks/useOptimizelyScrollDepth/index.test.jsx:64]: As above.

File: src/app/hooks/useTrackingToggle/index.test.jsx (hunk 20 above)
Comment (amoore108) [reply src/app/hooks/useTrackingToggle/index.test.jsx:13]: There is a note here on how to pass props when using `wrapper` https://testing-library.com/docs/react-testing-library/api/#renderhook-options-initialprops

File: src/app/legacy/components/MediaPlayer/Guidance/index.jsx (hunk 21 above)
Comment (amoore108) [reply src/app/legacy/components/MediaPlayer/Guidance/index.jsx]: I'll mark them as `''` by default to satisfy TS for the TSX components that pass in these values as strings.

File: src/app/legacy/components/MediaPlayer/Placeholder/index.jsx (hunk 22 above)
Comment (amoore108) [reply src/app/legacy/components/MediaPlayer/Placeholder/index.jsx]: Good spot, will change back.

File: src/app/legacy/components/OptimoPromos/Content/index.jsx (hunk 3 above)
Comment (amoore108) [reply src/app/legacy/components/OptimoPromos/Content/index.jsx:9]: This needs to be a string type as the TSX components using this pass in string values. Unless typed with TS types/interfaces, we can't declare multiple types for default values.

It should retain the same falsy default behaviour as null though.

File: src/app/legacy/components/Promo/timestamp.jsx (hunk 5 above)
Comment (amoore108) [reply src/app/legacy/components/Promo/timestamp.jsx:9]: Same as above, we can only specify the 1 type the default value can be. TSX components using this JSX component pass in string values, so TS doesn't like the type mismatch.

Should retain the same behaviour as a null though.

File: src/app/legacy/containers/ArticleTimestamp/index.jsx (hunk 24 above)
Comment (amoore108) [reply src/app/legacy/containers/ArticleTimestamp/index.jsx:22]: This needs set to a default value to satisfy the TS components that use this component. It detects that `className` is a prop, but without a default it will complain and ask you to provide a value for every implementation of the component.

File: src/app/legacy/containers/Brand/index.jsx (hunk 25 above)
Comment (amoore108) [reply src/app/legacy/containers/Brand/index.jsx:21]: See https://github.com/bbc/simorgh/pull/11612#discussion_r1596810735

File: src/app/legacy/containers/BulletedList/index.jsx (hunk 26 above)
Comment (amoore108) [reply src/app/legacy/containers/BulletedList/index.jsx:23]: As above, it needs to be a string to satisfy the TS compiler as some implementations of this component pass in classname as a string.

File: src/app/legacy/containers/PageHandlers/withPageWrapper/index.jsx (hunk 27 above)
Comment (amoore108) [reply src/app/legacy/containers/PageHandlers/withPageWrapper/index.jsx:19]: Will add

File: src/app/legacy/containers/WebVitals/index.jsx (hunk 28 above)
Comment (amoore108) [reply src/app/legacy/containers/WebVitals/index.jsx:52]: Shouldn't be required as it will be `null/undefined` anyway if not supplied.

File: src/app/legacy/psammead/psammead-heading-index/src/index.jsx (hunk 29 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-heading-index/src/index.jsx:12]: These are Emotion styled components, so I'm not sure of the best way to set ES6 default values on these. React doesn't seem to mind these even with version `18.3.x`

File: src/app/legacy/psammead/psammead-headings/src/index.jsx (hunk 30 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-headings/src/index.jsx:42]: As above.

File: src/app/legacy/psammead/psammead-story-promo-list/src/index.jsx (hunk 32 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-story-promo-list/src/index.jsx:32]: As above, same as the other Emotion styled components.

File: src/app/legacy/psammead/psammead-timestamp/src/index.jsx (hunk 33 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-timestamp/src/index.jsx:34]: As previously mentioned, TS will complain in TSX components importing JSX if the values supplied mismatch.

File: src/app/pages/ArticlePage/ArticlePage.jsx (hunk 35 above)
Comment (amoore108) [reply src/app/pages/ArticlePage/ArticlePage.jsx:280]: Can't see this being used anywhere in the component, so I removed it entirely.

File: src/app/components/MostRead/Canonical/LastUpdated/index.tsx (hunk 16 above)
Comment (amoore108) [reply src/app/components/MostRead/Canonical/LastUpdated/index.tsx:33]: This is a mismatch between using JSX components in TSX. `altCalendar` needs to be set to basically an `any` type in the JSX component (by not giving it a default value) so that TSX components can pass in either a `null` or altCalendar function.

Its not great, but its just a weirdness of compatibility between JSX and TSX types.

Comment (amoore108) [reply src/app/components/MostRead/Canonical/LastUpdated/index.tsx:33]: This is a disadvantage of removing propTypes/defaultProps. TypeScript does have compatibility with both and could more correctly infer types from the propTypes/defaultProps, but we need to get rid of them at some stage, so its a bit of a trade off until we move all components to TypeScript (at some stage!!)

File: src/app/lib/utilities/srcSet/index.js (hunk 34 above)
Comment (amoore108) [reply src/app/lib/utilities/srcSet/index.js:34]: Similar typing issue to the other `null` ones, whereby some TS components can supply a `string | undefined` rather than a `string | null`.

This shouldn't affect anything, but I agree is a weird oddity.

**Comment by amoore108** [comment]
> One question / concern - if we were to ever convert any of these components to Typescrpt, we don't have PropTypes to tell us if fields are required or not.
> 
> We will also need to update the Migrate Legacy Component doc to remove any mention of propTypes (if we haven't already done so)

Thats correct, we won't have a reference to the types that they can accept anymore. This is disadvantage of doing it this way since we have a hybrid of JS and TS. I guess refactoring components to TS will tighten up an loose types over time, but I can't think of any other ways other than something like JSDoc, which would be a huge undertaking.

I will remove the Readme references.

File: src/app/legacy/psammead/psammead-image/src/index.amp.jsx (hunk 31 above)
Comment (amoore108) [reply src/app/legacy/psammead/psammead-image/src/index.amp.jsx:12]: Not sure I follow 😬

**Comment by amoore108** [comment]
> Copypasta from Slack:
> 
> Nice! Thank you.
> 
> For the styled components issue with defaultProps, since React isn’t raising flags about their usage currently, it might be safe to leave them as is in the short term. However, for long-term maintenance and consistency, it might be worthwhile to schedule a gradual rewrite of these components into a supported pattern, possibly for when we tackle other refactoring tasks or feature upgrades.
> 
> Would it be worthwhile to add additional tests focusing on targeted components where defaultProps were replaced with default parameters to ensure there are no edge cases we're missing where the behaviour could diff?

Thanks, that makes sense. I think we're safe to keep the `defaultProps` on those components. They are more scarce than other components, so not like we'll have them littered everywhere.

Not really sure if its worth adding more tests. The current ones, along with snapshots/Chromatic should highlight regressions already, so I don't think adding more will help much? Happy to be wrong though!

**Review by andrewscfc — APPROVED** [review]
Thanks for the clarifications and changes, all looks good
